tasks/build.py

```diff
@@ -1,30 +1,113 @@
+import json
 import os
 import subprocess
+import tempfile
 #import shutil
 
 from connections import github
-from tools.logging import log
+from tools import config
+from pyghee.utils import log
 
-JOBS_DIR = os.path.join(os.getenv('HOME'), 'jobs')
-EASYSTACK_PATH = os.path.join('2021-12', '03_workflow', 'yaml-files', 'softwarelist.yaml')
-SCRIPTS_DIR = os.path.join(os.getenv('HOME'), 'git', 'hackathons', '2021-12', '03_workflow', 'scripts')
 
+def build_easystack_from_pr(pr, event_info):
+    # retrieving some settings from 'app.cfg' in bot directory
+    # [buildenv]
+    jobs_base_dir = config.get_section('buildenv').get('jobs_base_dir')
+    log("jobs_base_dir '%s'" % (jobs_base_dir))
+    scripts_dir = config.get_section('buildenv').get('scripts_dir')
+    log("scripts_dir '%s'" % (scripts_dir))
+    local_tmp = config.get_section('buildenv').get('local_tmp')
+    log("local_tmp '%s'" % (local_tmp))
+    build_job_script = config.get_section('buildenv').get('build_job_script')
+    log("build_job_script '%s'" % (build_job_script))
+    submit_command = config.get_section('buildenv').get('submit_command')
+    log("submit_command '%s'" % (submit_command))
 
-def build_easystack_from_pr(pr, request):
-    event_id = request.headers['X-Github-Delivery']
-    jobdir = os.path.join(JOBS_DIR, event_id)
-    os.makedirs(jobdir)
-    #shutil.copytree(SCRIPTS_DIR, jobdir)
-    # TODO: checkout the branch that belongs to the PR
+    # [softwaretargets]
+    # software_file currently not being used (may be later to limit what files will be processed)
+    software_file = config.get_section('softwaretargets').get('software_file')
+    log("software file '%s'" % (software_file))
+    whitelist_versions = json.loads(config.get_section('softwaretargets').get('whitelist_versions'))
+    log("whitelisted versions '%s'" % (json.dumps(whitelist_versions)))
+
+    # [architecturetargets]
+    arch_target_map = json.loads(config.get_section('architecturetargets').get('arch_target_map'))
+    log("arch target map '%s'" % (json.dumps(arch_target_map)))
+
+    # get event id and create main directory for event and a temporary dir
+    # inside that (the latter uses mktemp to create a new job every time the
+    # event is being redelivered, i.e., useful while developing)
+    event_id = event_info['id']
+    jobs_dir = os.path.join(jobs_base_dir, event_id)
```

Comment:
There should be a top directory for this specific PR here to give this some more structure, i.e. `pr<id>/jobs/<event_id>`, ...

Maybe we also need another directory level on top, per month (`2022.06/pr<id>/...`), so we can easily tell which old stuff we can remove or archive, and to avoid having lots of subdirectories

Not terribly important for now, can be changed later.